index/scorch/merge.go

```diff
@@ -263,6 +263,17 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
 	}
 
 	for _, notification := range notifications {
+		// update the idex snapshot reference counts for all
+		// the merge introductions which are already done.
+		if len(notification) == cap(notification) {
```

Comment:
Hi -- not sure if I understand the big picture here on this change yet, but looking at this from a very narrow-minded way (e.g., looking at the code through a mechanistic "paper towel roll" point of view)...

This kind of len(ch) == cap(ch) feels like an uncommon and perhaps even a frowned-upon pattern in golang?  Wondering, instead, if the notification channel is full then the reading from this channel might be still the right thing to do?

Reply:
hm..i don't think I am able to follow you..

> Wondering, instead, if the notification channel is full then the reading from this channel might be still the right thing to do?

"len(notification) == cap(notification)" is my attempt at checking whether notification channel is full or not.